core/server-core/src/services/app.ts

```diff
@@ -127,6 +136,80 @@ export class AppService {
     return app;
   }
 
+  async list(): Promise<DBApp[]> {
+    return await this.appDAO.list();
+  }
+
+  async create(
+    data: Omit<CreateUpdateAppData, "client_secret_hash"> & {
+      client_secret: string;
+    },
+    { actor, message }: AuditParams = {},
+  ): Promise<DBApp> {
+    const { client_secret, ...rest } = data;
```

Comment:
might be better if this is generated on the server side and we return it if a rotation is requested